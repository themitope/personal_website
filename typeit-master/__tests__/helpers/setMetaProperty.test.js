import setMetaProperty from "../../src/helpers/setMetaProperty";

test("Sets property on meta object of each item.", () => {
  let items = [
    ["action1", null, {}],
    ["action2", null, {}],
    ["action3", null, {}]
  ];

  let result = setMetaProperty(items, { freezeCursor: true });
  expect(result).toMatchSnapshot();
});

test("Sets property on meta object of each item even when meta doesn't yet exist.", () => {
  let items = [
    ["action1", null],
    ["action2", null, {}],
    ["action3", null]
  ];

  let result = setMetaProperty(items, { freezeCursor: true });
  expect(result).toMatchSnapshot();
});

test("Sets property on meta object of each item even when meta doesn't yet exist.", () => {
  let items = [
    ["action1", null],
    ["action2", null, {}],
    ["action3", null]
  ];

  let result = setMetaProperty(items, { freezeCursor: true });
  expect(result).toMatchSnapshot();
});

test("Sets property on meta object of single item.", () => {
  let result = setMetaProperty(["action", null], { freezeCursor: true });
  expect(result).toEqual(["action", null, { freezeCursor: true }]);
});
