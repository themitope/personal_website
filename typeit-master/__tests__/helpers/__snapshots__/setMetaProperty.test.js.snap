// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets property on meta object of each item even when meta doesn't yet exist. 1`] = `
Array [
  Array [
    "action1",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
  Array [
    "action2",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
  Array [
    "action3",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
]
`;

exports[`Sets property on meta object of each item even when meta doesn't yet exist. 2`] = `
Array [
  Array [
    "action1",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
  Array [
    "action2",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
  Array [
    "action3",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
]
`;

exports[`Sets property on meta object of each item. 1`] = `
Array [
  Array [
    "action1",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
  Array [
    "action2",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
  Array [
    "action3",
    null,
    Object {
      "freezeCursor": true,
    },
  ],
]
`;
